File: baseLib/src/main/java/me/ycdev/android/lib/common/async/HandlerExecutor.java
package me.ycdev.android.lib.common.async;

import android.os.Handler;
import android.os.Looper;
import android.support.annotation.NonNull;

import java.util.List;

@SuppressWarnings({"unused", "WeakerAccess"})
public class HandlerExecutor implements ITaskExecutor {
    private Handler mTaskHandler;

    public HandlerExecutor(@NonNull Looper looper) {
        mTaskHandler = new Handler(looper);
    }

    @Override
    public void postTasks(@NonNull List<Runnable> tasks) {
        for (Runnable task : tasks) {
            mTaskHandler.post(task);
        }
    }

    @Override
    public void clearTasks() {
        mTaskHandler.removeCallbacksAndMessages(null);
    }
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/async/HandlerExecutor.kt
package me.ycdev.android.lib.common.async

import android.os.Handler
import android.os.HandlerThread
import android.os.Looper

open class HandlerExecutor(looper: Looper) : ITaskExecutor {
    private val taskHandler: Handler = Handler(looper)

    override fun postTasks(tasks: List<Runnable>) {
        for (task in tasks) {
            taskHandler.post(task)
        }
    }

    override fun clearTasks() {
        taskHandler.removeCallbacksAndMessages(null)
    }

    companion object {
        fun withMainLooper(): HandlerExecutor {
            return HandlerExecutor(Looper.getMainLooper())
        }

        fun withHandlerThread(name: String): HandlerExecutor {
            val thread = HandlerThread(name)
            thread.start()
            return HandlerExecutor(thread.looper)
        }
    }
}
